endpoints/getting-started/client/main.go

```diff
@@ -26,150 +23,103 @@ import (
+	host                = flag.String("host", "", "The API host. Required.")
+	audience            = flag.String("audience", "", "The audience for the JWT. equired")
+	serviceAccountFile  = flag.String("service-account-file", "", "Path to service account JSON file. Required.")
+	serviceAccountEmail = flag.String("service-account-email", "", "Path email associated with the service account. Required.")
 )
 
 func main() {
 	flag.Parse()
 
-	if *apiKey == "" || *host == "" {
-		flag.PrintDefaults()
-		os.Exit(1)
-	}
-	if *serviceAccount == "" && *echo == "" {
-		fmt.Fprint(os.Stderr, "Provide one of -echo or -service-account.")
-		flag.PrintDefaults()
-		os.Exit(1)
-	}
-	if *serviceAccount != "" && *echo != "" {
-		fmt.Fprint(os.Stderr, "Provide only one of -echo or -service-account.")
-		flag.PrintDefaults()
+	if *audience == "" || *host == "" || *serviceAccountFile == "" || *serviceAccountEmail == "" {
+		fmt.Println("requires: --host, --audience, --service-account-file, --service-account-email")
 		os.Exit(1)
 	}
-
-	var resp *http.Response
-	var err error
-	if *echo != "" {
-		resp, err = doEcho()
-	} else if *serviceAccount != "" {
-		resp, err = doJWT()
-	}
+	jwt, err := generateJWT(*serviceAccountFile, *serviceAccountEmail, *audience, 3600)
 	if err != nil {
 		log.Fatal(err)
 	}
-	b, err := httputil.DumpResponse(resp, true)
+	fmt.Println(jwt)
+	resp, err := makeJWTRequest(jwt, *host)
 	if err != nil {
 		log.Fatal(err)
 	}
-	os.Stdout.Write(b)
+	fmt.Println(resp)
 }
 
-// doEcho performs an authenticated echo request using an API key.
-func doEcho() (*http.Response, error) {
-	msg := map[string]string{
-		"message": *echo,
+// [START endpoints_generate_jwt_sa]
+
+// generateJWT creates a signed JSON Web Token using a Google API Service Account.
+func generateJWT(saKeyfile, saEmail, audience string, expiryLength int64) (string, error) {
+	now := time.Now().Unix()
+
+	// build payload
+	jwt := &jws.ClaimSet{
+		Iat: now,
+		// expires after 'expiraryLength' seconds.
+		Exp: now + expiryLength,
+		// Iss must match 'issuer' in the security configuration in your
+		// swagger spec (e.g. service account email). It can be any string.
+		Iss: saEmail,
+		// Aud must be either your Endpoints service name, or match the value
+		// specified as the 'x-google-audience' in the OpenAPI document.
+		Aud: audience,
+		// Sub and Email should match the service account's email address
+		Sub:           saEmail,
+		PrivateClaims: map[string]interface{}{"email": saEmail},
 	}
-	var buf bytes.Buffer
-	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return nil, err
+	jwsHeader := &jws.Header{
+		Algorithm: "RS256",
+		Typ:       "JWT",
 	}
-	return http.Post(*host+"/echo?key="+*apiKey, "application/json", &buf)
-}
 
-// doJWT performs an authenticated request using the credentials in the service account file.
-func doJWT() (*http.Response, error) {
-	sa, err := ioutil.ReadFile(*serviceAccount)
+	// sign with keyfile
```

Comment:
Either delete or capitalize + add a period. If commenting, might want more than this below. It's hard to understand what's happening in each step.